transport/http/outbound.go

```diff
@@ -386,16 +390,24 @@ func (o *Outbound) withCoreHeaders(req *http.Request, treq *transport.Request, t
 		req.Header.Set(EncodingHeader, encoding)
 	}
 
+	req.Header.Set(AcceptsBothResponseErrorHeader, AcceptTrue)
+
 	return req
 }
 
-func getYARPCErrorFromResponse(response *http.Response) error {
-	contents, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return yarpcerrors.Newf(yarpcerrors.CodeInternal, err.Error())
-	}
-	if err := response.Body.Close(); err != nil {
-		return yarpcerrors.Newf(yarpcerrors.CodeInternal, err.Error())
+func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool) error {
```

Comment:
As opposed to weaving the boolean parameter through this function, it could increase readability to make two distinct functions here:

`getYARPCErrorFromResponseHeader`
`getYARPCErrorFromResponseBody`

And have each of those functions call into a shared `getYARPCErrorStatusCode` function.

Reply:
I tried to do this out but it seems worse to me

Reply:
Consider using an enum to avoid the boolean trap.

Reply:
An enum? I really think this solution is fine, and we don't need to optimize it much more - adding another type will just get more complicated.